modules/programs/btop.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.btop;
+
+  formatLine = n: v:
+    let
+      formatValue = v:
+        if builtins.isBool v then
+          (if v then "True" else "False")
+        else if builtins.isString v then
+          ''"${v}"''
+        else
+          toString v;
+    in "${n} = ${formatValue v}";
+in {
+  meta.maintainers = [ hm.maintainers.GaetanLepage ];
+
+  options.programs.btop = {
+    enable = mkEnableOption "btop";
+
+    settings = mkOption {
+      type = with types; attrsOf (oneOf [ bool float int str ]);
+      default = { };
+      description = ''
+        Options to add to <filename>btop.conf</filename> file.
+        <link xlink:href="https://github.com/aristocratos/btop#configurability"/>
+        for options.
+      '';
+      example = {
+        color_theme = "Default";
+        theme_background = false;
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra lines added to <filename>btop.conf</filename> file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.btop ];
```

Comment:
Can you add a configurable `package` option and use `cfg.package` here?

`package = mkPackageOption pkgs "btop" {}`

Reply:
Nice, I didn't know a `mkPackageOption` was added 🙂

Reply:
done :)

Reply:
I updated the PR to use `mkPackageOption`, hopefully it'll work OK. Note, you need to use Nixpkgs unstable.